### PR TITLE
hack/build: Pin to RHCOS 400.7.20190306.0

### PR DESCRIPTION
**Pull request by wking**
@smarterclayton pushed release:4.0.0-0.nightly-2019-03-04-234414 to quay.io/openshift-release-dev/ocp-release:4.0.0-0.7.  Extracting the associated RHCOS build:

```console
$ oc adm release info --pullspecs quay.io/openshift-release-dev/ocp-release:4.0.0-0.7 | grep machine-os-content
  machine-os-content                            quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:399582f711226ab1a0e76d8928ec55436dea9f8dc60976c10790d308b9d92181
$ oc image info quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:399582f711226ab1a0e76d8928ec55436dea9f8dc60976c10790d308b9d92181 | grep version
            version=47.330
```

I'd like to be pulling that version out of the release image at install-time (#1286), but that PR isn't green yet, so hard-coding here is a quick fix.

CC @abhinavdahiya, @cgwalters

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 set -ex
 
+RHCOS_BUILD_NAME="${RELEASE_BUILD_NAME:-47.330}"
```
Comment (abhinavdahiya) [inline hack/build.sh]: `/s/RELEASE_BUILD_NAME/RHCOS_BUILD_NAME` ?

**Comment by cgwalters** [comment]
Very short term this is OK but soon we will need a newer version to get authenticated pivots to work.

File: hack/build.sh (hunk 1 above)
Comment (wking) [reply hack/build.sh]: > `/s/RELEASE_BUILD_NAME/RHCOS_BUILD_NAME` ?

Oops.  I have been making that mistake in released images for a while :/.  Fixed with 6f7c83b2b -> bd88157ab.

**Comment by wking** [comment]
> Very short term this is OK but soon we will need a newer version to get authenticated pivots to work.

Bumped to 400.7.20190306.0 with 6f7c83b2b -> bd88157ab.

**Comment by cgwalters** [comment]
This LGTM but I am going to rebase https://github.com/openshift/installer/pull/1402 on top and also add an environment variable to select the build so we can conveniently test new versions without rebuilding the installer.

File: hack/build.sh [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 set -ex
 
+RHCOS_BUILD_NAME="${RHCOS_BUILD_NAME:-400.7.20190306.0}"
```
Comment (ashcrow) [inline hack/build.sh]: I don't have a problem with this but note that the version referenced isn't an official cut.

Comment (wking) [reply hack/build.sh]: > I don't have a problem with this but note that the version referenced isn't an official cut.

I don't care what we pin to, just tell me what you want ;)

Comment (ashcrow) [reply hack/build.sh]: This is fine. I just wanted to note that it wasn't an officially ID'd build for any specific cut.

**Comment by cgwalters** [comment]
:shipit: ?

File: hack/build.sh (hunk 2 above)
Comment (ashcrow) [reply hack/build.sh]: FWIW `400.7.20190312.0` now exists which includes a `pivot` update among other things. @cgwalters do you need that as part of the pin?

Comment (cgwalters) [reply hack/build.sh]: My tentative vote here is to ship this as is now and start building "mental muscle memory" or SOP for bumping it immediately after, but it's not a really strong opinion.

**Comment by ashcrow** [comment]
resource quota and replica set errors. Flakes?

**Comment by cgwalters** [comment]
/retest

Yeah we have a successful run with the same commit.

**Pull request merged**